vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/codec/json/JsonObjectCodec.java

```diff
@@ -187,13 +189,19 @@ protected void writeDateTime(BsonWriter writer, String name, Object value, Encod
   @Override
   protected Object readBinary(BsonReader reader, DecoderContext ctx) {
     final JsonObject result = new JsonObject();
-    result.put(BINARY_FIELD, reader.readBinaryData().getData());
+    BsonBinary tmpObj = reader.readBinaryData();
+    result.put(BINARY_FIELD, tmpObj.getData())
```

Comment:
It's not really a `tmpObj`?  I guess it's data?  A document?  Something like that.